File: util/wechat/src/main/java/cn/bestwu/summer/util/wechat/config/WechatAppConfiguration.java
package cn.bestwu.summer.util.wechat.config;

import cn.bestwu.summer.util.wechat.controller.WechatAppCallbackController;
import cn.bestwu.summer.util.wechat.support.IWechatService;
import com.riversoft.weixin.app.base.AppSetting;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@ConditionalOnClass(AppSetting.class)
@ConditionalOnProperty(prefix = "wechat.mini", name = "appId")
@Configuration
@EnableConfigurationProperties({WechatMiniAppProperties.class})
public class WechatAppConfiguration {

  private final Logger log = LoggerFactory.getLogger(WechatAppConfiguration.class);
  private final WechatMiniAppProperties wechatMiniAppProperties;

  public WechatAppConfiguration(WechatMiniAppProperties wechatMiniAppProperties) {
    this.wechatMiniAppProperties = wechatMiniAppProperties;
  }


  @Bean
  public WechatAppCallbackController wechatAppCallbackController(IWechatService wechatService) {
    return new WechatAppCallbackController(wechatService, wechatMiniAppProperties);
  }


}

File: tools/weixin/src/main/kotlin/top/bettercode/summer/tools/weixin/config/WechatAppConfiguration.kt
package top.bettercode.summer.tools.weixin.config

import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import top.bettercode.summer.tools.weixin.controller.MiniprogramCallbackController
import top.bettercode.summer.tools.weixin.properties.IMiniprogramProperties
import top.bettercode.summer.tools.weixin.support.IWechatService
import top.bettercode.summer.tools.weixin.support.miniprogram.IMiniprogramClient
import top.bettercode.summer.tools.weixin.support.miniprogram.MiniprogramClient

@Configuration(proxyBeanMethods = false)
@ConditionalOnBean(IMiniprogramProperties::class)
class WechatAppConfiguration(private val miniprogramProperties: IMiniprogramProperties) {

    @Bean
    fun miniprogramClient(): IMiniprogramClient {
        return MiniprogramClient(miniprogramProperties)
    }

    @ConditionalOnBean(IWechatService::class)
    @ConditionalOnWebApplication
    @Bean
    fun miniprogramCallbackController(
        wechatService: IWechatService,
        miniprogramClient: IMiniprogramClient
    ): MiniprogramCallbackController {
        return MiniprogramCallbackController(wechatService, miniprogramClient)
    }

}